geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,55 +8,234 @@
+      None
     } else {
-      // TODO: Synchronize Blobstore creation
-      // https://geomesa.atlassian.net/browse/GEOMESA-985
-      abs = new AccumuloBlobStore(ds)
-      Ok()
+      Some(new AccumuloBlobStore(ds))
     }
   }
 
-  get("/:id") {
-    val id = params("id")
-    logger.debug(s"In ID method, trying to retrieve id $id")
+  /**
+    * Override of the requestPath function, may need to match on type of request if wrapped
+    * depending on deployment environment
+    *
+    * @param request
+    * @return
+    */
+  override def requestPath(implicit request: HttpServletRequest): String = request match {
+    case r: HttpServletRequestWrapper =>
+      r.getMethod.toLowerCase(Locale.ENGLISH) match {
+        case "post" =>
+          if (r.getServletPath == "/blob") {
+            s"/blob${r.getPathInfo}"
+          } else {
+            r.getPathInfo
+          }
+        case "delete" =>
+          r.getPathInfo.replace(s"/$root", "")
+        case _        =>
+          r.getPathInfo
+      }
+    case _ => super.requestPath
+  }
 
-    if (abs == null) {
-      NotFound(reason = "AccumuloBlobStore is not initialized.")
+  // TODO: Revisit configuration and persistence of configuration.
+  // https://geomesa.atlassian.net/browse/GEOMESA-958
+  /**
+    * Registers a data store, making it available for later use
+    */
+  post("/ds/:alias/?") {
+    logger.debug("Attempting to register accumulo connection in Blob Store")
+    val dsParams = datastoreParams
+    val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+    if (ds == null) {
+      BadRequest(reason = "Could not load data store using the provided parameters.")
     } else {
-      val (returnBytes, filename) = abs.get(id)
-      if (returnBytes == null) {
-        NotFound(reason = s"Unknown ID $id")
-      } else {
-        contentType = "application/octet-stream"
-        response.setHeader("Content-Disposition", "attachment; filename=" + filename)
-
-        Ok(returnBytes)
+      val alias = params("alias")
+      val prefix = keyFor(alias)
+      val toPersist = dsParams.map { case (k, v) => keyFor(alias, k) -> v }
+      try {
+        persistence.removeAll(persistence.keys(prefix).toSeq)
+        persistence.persistAll(toPersist)
+        blobStores.put(alias, new AccumuloBlobStore(ds))
+        Ok()
+      } catch {
+        case e: Exception => handleError(s"Error persisting data store '$alias':", e)
       }
     }
   }
+
+  /**
+    * Retrieve an existing data store
+    */
+  get("/ds/:alias/?") {
+    try {
+      getPersistedDataStore(params("alias"))
+    } catch {
+      case e: Exception => handleError(s"Error reading data store:", e)
+    }
+  }
+
+  /**
+    * Remove the reference to an existing data store
+    */
+  delete("/ds/:alias/?") {
+    val alias = params("alias")
+    val prefix = keyFor(alias)
+    try {
+      persistence.removeAll(persistence.keys(prefix).toSeq)
```

Comment:
no - this method doesn't really delete anything - just removes the reference to it in the 'registered' aliases. So in your case, you also need to remove the reference in the blobstore map.